op-service/eth/blob.go

```diff
@@ -0,0 +1,133 @@
+package eth
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"reflect"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto/kzg4844"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+const (
+	BlobSize        = 4096 * 32
+	MaxBlobDataSize = 4096*31 - 4
+)
+
+type Blob [BlobSize]byte
+
+func (b *Blob) KZGBlob() *kzg4844.Blob {
+	return (*kzg4844.Blob)(b)
+}
+
+func (b *Blob) UnmarshalJSON(text []byte) error {
+	return hexutil.UnmarshalFixedJSON(reflect.TypeOf(b), text, b[:])
+}
+
+func (b *Blob) UnmarshalText(text []byte) error {
+	return hexutil.UnmarshalFixedText("Bytes32", text, b[:])
+}
+
+func (b *Blob) MarshalText() ([]byte, error) {
+	return hexutil.Bytes(b[:]).MarshalText()
+}
+
+func (b *Blob) String() string {
+	return hexutil.Encode(b[:])
+}
+
+// TerminalString implements log.TerminalStringer, formatting a string for console
+// output during logging.
+func (b *Blob) TerminalString() string {
+	return fmt.Sprintf("%x..%x", b[:3], b[BlobSize-3:])
+}
+
+func (b *Blob) ComputeKZGCommitment() (kzg4844.Commitment, error) {
+	return kzg4844.BlobToCommitment(*b.KZGBlob())
+}
+
+// KZGToVersionedHash computes the "blob hash" (a.k.a. versioned-hash) of a blob-commitment, as used in a blob-tx.
+// We implement it here because it is unfortunately not (currently) exposed by geth.
+func KZGToVersionedHash(commitment kzg4844.Commitment) (out common.Hash) {
+	// EIP-4844 spec:
+	//	def kzg_to_versioned_hash(commitment: KZGCommitment) -> VersionedHash:
+	//		return VERSIONED_HASH_VERSION_KZG + sha256(commitment)[1:]
+	h := sha256.New()
+	h.Write(commitment[:])
+	_ = h.Sum(out[:0])
+	out[0] = params.BlobTxHashVersion
+	return out
+}
```

Comment:
We never use this so can remove it (at least for now).

Reply:
Actually we don't use anything from this file except in its tests.  So we can either remove them both or if we want to verify the RPC blob responses we'd need to add something into the `sources/types.go` `rpcBlock.verify()` method.

Given we're not using the blob data yet, I'd actually be tempted to not verify it at this point. Even if the RPC returns invalid blob data it won't affect the derivation result - we just need to ensure the new fields are added into the block header so we compute the correct expected hash. @trianglesphere @protolambda any preferences on this?

Reply:
How expensive of an operation is it to verify them? Is the risk having a bug in the verification that halts syncing?

Reply:
No need to verify/load the commitments of the individual blob txs during block-header verification; it's not part of the header. When we utilize the full blob for as DA later, we verify it as part of the blob-fetching. For now I think it's fine to include this helper function; it's pretty basic, and geth does not expose it (at least, last time I checked), so it's nice to have. Also fine to defer to a later PR, if that's preferred.

Reply:
Yeah I'm happy either way but would lean towards leaving it for a future PR where it's actually used so it can be better evaluated in context.